spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/IcebergAdaptor.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.connector.iceberg;
+
+import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.spark.connector.GravitinoCatalogAdaptor;
+import com.datastrato.gravitino.spark.connector.GravitinoSparkConfig;
+import com.datastrato.gravitino.spark.connector.PropertiesConverter;
+import com.datastrato.gravitino.spark.connector.table.SparkBaseTable;
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/** IcebergAdaptor provides specific operations for Iceberg Catalog to adapt to GravitinoCatalog. */
+public class IcebergAdaptor implements GravitinoCatalogAdaptor {
+
+  @Override
+  public PropertiesConverter getPropertiesConverter() {
+    return new IcebergPropertiesConverter();
+  }
+
+  @Override
+  public SparkBaseTable createSparkTable(
+      Identifier identifier,
+      Table gravitinoTable,
+      TableCatalog sparkCatalog,
+      PropertiesConverter propertiesConverter) {
+    return new SparkIcebergTable(identifier, gravitinoTable, sparkCatalog, propertiesConverter);
+  }
+
+  @Override
+  public TableCatalog createAndInitSparkCatalog(
+      String name, CaseInsensitiveStringMap options, Map<String, String> properties) {
+    Preconditions.checkArgument(
+        properties != null, "Iceberg Catalog properties should not be null");
+
+    String catalogBackend = properties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_BACKEND);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(catalogBackend), "Iceberg Catalog backend should not be empty.");
+
+    HashMap<String, String> all = new HashMap<>(options);
+
+    switch (catalogBackend.toLowerCase(Locale.ENGLISH)) {
+      case GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_BACKEND_HIVE:
+        String metastoreUri = properties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI);
```

Comment:
the function is too long, could you extract to simple function for different catalog backend?

Reply:
updated.